x/fileurl.go

```diff
@@ -0,0 +1,109 @@
+package x
+
+import (
+	"net/url"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
+
+	"github.com/ory/x/logrusx"
+)
+
+// winPathRegex is a regex for [DRIVE-LETTER]:
+var winPathRegex = regexp.MustCompile("^[A-Za-z]:.*")
+
+// GetURLFilePath returns the path of a URL that is compatible with the runtime os filesystem
+func GetURLFilePath(u *url.URL) string {
+	if u == nil {
+		return ""
+	}
+	if !(u.Scheme == "file" || u.Scheme == "") {
+		return u.Path
+	}
+
+	fPath := u.Path
+	if runtime.GOOS == "windows" {
+		if u.Host != "" {
+			// Make UNC Path
+			fPath = "\\\\" + u.Host + filepath.FromSlash(fPath)
+			return fPath
+		}
+		fPathTrimmed := strings.TrimLeft(fPath, "/")
+		if winPathRegex.MatchString(fPathTrimmed) {
+			// On Windows we should remove the initial path separator in case this
+			// is a normal path (for example: "\c:\" -> "c:\"")
+			fPath = fPathTrimmed
+		}
+	}
+	return filepath.FromSlash(fPath)
+}
+
+// ParseURL parses rawURL into a URL structure with special handling for file:// URLs
+func ParseURL(rawURL string) (*url.URL, error) {
+	lcRawURL := strings.ToLower(rawURL)
+	if strings.HasPrefix(lcRawURL, "file:///") {
+		return url.Parse(rawURL)
+	}
+
+	if strings.HasPrefix(lcRawURL, "file://") {
```

Comment:
I just realized you can avoid the `if` statement and use `strings.TrimPrefix(lcRawURL, "file://")`, sry :sweat_smile:

Reply:
I has the same though, but it turns out that we are comparing the lower case converted string but modifying the non converted string so unfortunately that won't work. Unless there is a way to TrimPrefix without case sensitivity?

Reply:
Ah no, you are actually trimming `rawURL`, nevermind.